git-webui/src/share/git-webui/webui/js/git-webui.js

```diff
@@ -3012,35 +3012,43 @@ webui.NewChangedFilesView = function(workspaceView) {
     }
 
     self.amend = function(message, details) {
-
         if (self.commitMsgEmpty()) {
-            webui.git_command(["add"].concat(selectedItems));
-            webui.git_command(['commit', '--amend', '--no-edit', '--'].concat(selectedItems), function(output) {
-                webui.showSuccess(output);
+            $.post("git-command-unstage",{command: "commit", flags: "--amend,--no-edit", files:JSON.stringify({files:selectedItems})}, function(output) {
+                if (output.includes("Error")) {
```

Comment:
Pretty sure this will break on IE. Would be better to indicate error through something other than output content.